sequence of bools in IDL doesn't work with libc++

## Required information

**Operating system:**
Ubuntu 18.04 LTS

**Cyclone version:**
- cyclonedds - master, [779c58](https://github.com/eclipse-cyclonedds/cyclonedds/commit/779c58206d6240a2cc405eece9a66bf70b97cd00)
- cyclonedds-cxx - master, 624da660d9d2c725555bd65bf5ff5c879dbee77b

**Observed result or behaviour:**

When a sequence of bools are used in the IDL, the templated streaming functions are not resolved correctly

**Detailed reproducer**

When using the following IDL file

```hpp
module test_msgs {
module msg {
module dds_ {
struct BoundedSequences_ {
sequence<boolean, 3> bool_values_;
};
};  // module dds_
};  // module msg
};  // module test_msgs
```
in the code as below:
```cpp
TEST(tooch, bound)
{
  using DDSType = test_msgs::msg::dds_::BoundedSequences_;
  dds::domain::DomainParticipant dp(0);
  dds::pub::Publisher pub(dp);
  dds::topic::Topic<DDSType> topic(dp, "test");
  dds::pub::DataWriter<DDSType> dw(pub, topic);
  DDSType msg{};
  std::vector<bool> vb{true, false, false, true};
  msg.bool_values_(vb);
  dw.write(msg);
}
```

Compile the code with `Clang` and it fails to resolve the templated streaming functions correctly with the following error

```sh
/home/sumanth.nirmal/tooch/new_idl/sequence_boolean.hpp:172:3: error: no matching function for call to 'move'
  move(streamer, instance.bool_values_()[i_1]);
  ^~~~
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/topic/datatopic.hpp:320:5: note: in instantiation of function template specialization 'org::eclipse::cyclonedds::core::cdr::move<org::eclipse::cyclonedds::core::cdr::basic_cdr_stream>' requested here
    move(str, msg);
    ^
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/topic/datatopic.hpp:510:4: note: in instantiation of function template specialization 'serdata_from_sample<test_msgs::msg::dds_::BoundedSequences_>' requested here
  &serdata_from_sample<T>,
   ^
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/topic/datatopic.hpp:538:25: note: in instantiation of static data member 'ddscxx_serdata<test_msgs::msg::dds_::BoundedSequences_>::ddscxx_serdata_ops' requested here
    &ddscxx_serdata<T>::ddscxx_serdata_ops,
                        ^
/home/sumanth.nirmal/tooch/new_idl/sequence_boolean.hpp:80:20: note: in instantiation of member function 'ddscxx_sertype<test_msgs::msg::dds_::BoundedSequences_>::ddscxx_sertype' requested here
    auto *st = new ddscxx_sertype<::test_msgs::msg::dds_::BoundedSequences_>();
                   ^
/home/sumanth.nirmal/tooch/new_idl/sequence_boolean.hpp:163:6: note: candidate function not viable: no known conversion from 'std::__1::vector<bool, std::__1::allocator<bool> >::const_reference' (aka '__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >') to 'const ::test_msgs::msg::dds_::BoundedSequences_' for 2nd argument
void move(T& streamer, const ::test_msgs::msg::dds_::BoundedSequences_& instance)
     ^
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/core/cdr/basic_cdr_ser.hpp:126:6: note: candidate template ignored: requirement 'std::is_arithmetic<__bit_const_reference<vector<bool, allocator<bool> > > >::value' was not satisfied [with T = std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >]
void move(basic_cdr_stream& str, const T& toincr)
     ^
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/core/cdr/basic_cdr_ser.hpp:209:6: note: candidate template ignored: requirement 'std::is_enum<__bit_const_reference<vector<bool, allocator<bool> > > >::value' was not satisfied [with T = std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >]
void move(basic_cdr_stream& str, const T& toincr) {
     ^
/usr/include/c++/v1/type_traits:2290:1: note: candidate function template not viable: requires single argument '__t', but 2 arguments were provided
move(_Tp&& __t) _NOEXCEPT
^
/usr/include/c++/v1/iterator:1477:48: note: candidate function template not viable: requires 3 arguments, but 2 were provided
    template <class _Ip, class _Op> friend _Op move(_Ip, _Ip, _Op);
                                               ^
/usr/include/c++/v1/__bit_reference:691:1: note: candidate function template not viable: requires 3 arguments, but 2 were provided
move(__bit_iterator<_Cp, _IsConst> __first, __bit_iterator<_Cp, _IsConst> __last, __bit_iterator<_Cp, false> __result)
^
/usr/include/c++/v1/deque:725:1: note: candidate function template not viable: requires at least 3 arguments, but 2 were provided
move(_RAIter __f,
^
/usr/include/c++/v1/deque:755:1: note: candidate function template not viable: requires 3 arguments, but 2 were provided
move(__deque_iterator<_V1, _P1, _R1, _M1, _D1, _B1> __f,
^
/usr/include/c++/v1/deque:783:1: note: candidate function template not viable: requires 3 arguments, but 2 were provided
move(__deque_iterator<_V1, _P1, _R1, _M1, _D1, _B1> __f,
^
In file included from /home/sumanth.nirmal/gc/apex_ws/src/apex_os/core/apexutils/test/test_apexutils.cpp:430:
/home/sumanth.nirmal/tooch/new_idl/sequence_boolean.hpp:133:3: error: no matching function for call to 'write'
  write(streamer, instance.bool_values_()[i_1]);
  ^~~~~
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/topic/datatopic.hpp:339:5: note: in instantiation of function template specialization 'org::eclipse::cyclonedds::core::cdr::write<org::eclipse::cyclonedds::core::cdr::basic_cdr_stream>' requested here
    write(str, msg);
    ^
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/topic/datatopic.hpp:510:4: note: in instantiation of function template specialization 'serdata_from_sample<test_msgs::msg::dds_::BoundedSequences_>' requested here
  &serdata_from_sample<T>,
   ^
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/topic/datatopic.hpp:538:25: note: in instantiation of static data member 'ddscxx_serdata<test_msgs::msg::dds_::BoundedSequences_>::ddscxx_serdata_ops' requested here
    &ddscxx_serdata<T>::ddscxx_serdata_ops,
                        ^
/home/sumanth.nirmal/tooch/new_idl/sequence_boolean.hpp:80:20: note: in instantiation of member function 'ddscxx_sertype<test_msgs::msg::dds_::BoundedSequences_>::ddscxx_sertype' requested here
    auto *st = new ddscxx_sertype<::test_msgs::msg::dds_::BoundedSequences_>();
                   ^
/home/sumanth.nirmal/tooch/new_idl/sequence_boolean.hpp:124:6: note: candidate function not viable: no known conversion from 'std::__1::vector<bool, std::__1::allocator<bool> >::const_reference' (aka '__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >') to 'const ::test_msgs::msg::dds_::BoundedSequences_' for 2nd argument
void write(T& streamer, const ::test_msgs::msg::dds_::BoundedSequences_& instance)
     ^
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/core/cdr/basic_cdr_ser.hpp:100:6: note: candidate template ignored: requirement 'std::is_arithmetic<__bit_const_reference<vector<bool, allocator<bool> > > >::value' was not satisfied [with T = std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >]
void write(basic_cdr_stream& str, const T& towrite)
     ^
/home/sumanth.nirmal/cyclone_iox_ws/cyclonedds-cxx/install/include/ddscxx/org/eclipse/cyclonedds/core/cdr/basic_cdr_ser.hpp:197:6: note: candidate template ignored: requirement 'std::is_enum<__bit_const_reference<vector<bool, allocator<bool> > > >::value' was not satisfied [with T = std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >]
void write(basic_cdr_stream& str, const T& towrite) {
     ^
```

The reason for this is that the const reference of boolean vector is not defined as bool in `libc++` though the standard seems to say so https://en.cppreference.com/w/cpp/container/vector_bool

**Expected result or behaviour:**

The sequence of bool type should also work with libc++

--- a/src/idlcxx/src/streamers.c
+++ b/src/idlcxx/src/streamers.c
@@ -285,6 +285,12 @@ write_basic_type_streaming_functions(
       "    %2$s(streamer, b);\n"
       "    %1$s = b;\n"
       "  }\n";
+
+    fmt =
+        "  {\n"
+        "    const bool b = %1$s;\n"
+        "    %2$s(streamer, b);\n"
+        "  }\n";
   }
 
   if ((loc.type & NORMAL_INSTANCE) &&

